js/apps/system/_admin/aardvark/APP/react/src/components/form/CreatableSingleSelectControl.tsx

```diff
@@ -0,0 +1,45 @@
+import { useField, useFormikContext } from "formik";
+import React from "react";
+import { Props as ReactSelectProps, PropsValue } from "react-select";
+import CreatableSingleSelect from "../select/CreatableSingleSelect";
+import { BaseFormControlProps, FormikFormControl } from "./FormikFormControl";
+
+type OptionType = {
+  label: string;
+  value: string;
+};
+export type InputControlProps = BaseFormControlProps & {
+  selectProps?: ReactSelectProps<OptionType, false>;
+};
+
+export const CreatableSingleSelectControl = (props: InputControlProps) => {
+  const { name, label, selectProps, ...rest } = props;
+  const [field, , helper] = useField(name);
+  const { isSubmitting } = useFormikContext();
+
+  let value = selectProps?.options?.find(option => {
+    return (option as OptionType).value === field.value;
+  }) as PropsValue<OptionType>;
+
+  // this is when a value is newly created
+  if (!selectProps?.options && field.value) {
```

Comment:
Isn't this the same as the value above not being set yet?

This could be:
```suggestion
  if (!value && field.value) {
```

Reply:
refactored this whole logic into a separate function